editor/editor_file_system.cpp

```diff
@@ -1948,6 +1948,16 @@ EditorFileSystemDirectory *EditorFileSystem::find_file(const String &p_file, int
 	return fs;
 }
 
+ResourceUID::ID EditorFileSystem::get_file_uid(const String &p_path) const {
+	int file_idx;
+	EditorFileSystemDirectory *directory = find_file(p_path, &file_idx);
+
+	if (!directory) {
+		return ResourceUID::INVALID_ID;
+	}
+	return directory->files[file_idx]->uid;
```

Comment:
Is there any risk of `directory->files[file_idx]` being `nullptr`?

Reply:
It's assigned here:
https://github.com/godotengine/godot/blob/296de7da830fa03425edf544427543087817e649/editor/editor_file_system.cpp#L1909-L1914
So the index and file are always valid at least at the point where it's assigned.